File: src/main/java/nz/co/searchwellington/views/GeocodeToPlaceMapper.java
package nz.co.searchwellington.views;

import nz.co.searchwellington.model.Geocode;

import org.springframework.stereotype.Component;

import uk.co.eelpieconsulting.common.geo.model.LatLong;
import uk.co.eelpieconsulting.common.geo.model.OsmId;
import uk.co.eelpieconsulting.common.geo.model.OsmType;
import uk.co.eelpieconsulting.common.geo.model.Place;

@Component
public class GeocodeToPlaceMapper {
	
	public Place mapGeocodeToPlace(final Geocode contentItemGeocode) {
		LatLong latLong = null;
		if (contentItemGeocode.getLatitude() != null && contentItemGeocode.getLongitude() != null) {
			latLong = new LatLong(contentItemGeocode.getLatitude(), contentItemGeocode.getLongitude());
		}
		OsmId osmId = null;
		if (contentItemGeocode.getOsmId() != null && contentItemGeocode.getOsmType() != null) {
			osmId = new OsmId(contentItemGeocode.getOsmId(), OsmType.valueOf(contentItemGeocode.getOsmType()));
		}
		return new Place(contentItemGeocode.getDisplayName(), latLong, osmId);
	}

}

File: src/main/scala/nz/co/searchwellington/views/GeocodeToPlaceMapper.scala
package nz.co.searchwellington.views

import nz.co.searchwellington.model.Geocode
import nz.co.searchwellington.model.frontend.{LatLong, OSMId, Place}
import org.springframework.stereotype.Component

@Component class GeocodeToPlaceMapper {

  def mapGeocodeToPlace(geocode: Geocode): Place = {
    val latLong = geocode.latitude.flatMap { lat =>
      geocode.longitude.map { lon =>
        LatLong(lat, lon)
      }
    }

    val osmId = geocode.osmId.flatMap { oid =>
      geocode.osmType.map { ot =>
        OSMId(oid, ot)
      }
    }

    Place(geocode.getDisplayName, latLong, osmId)
  }

}